cmake/pybind11_add_holohub_module.cmake

```diff
@@ -49,6 +49,15 @@ function(pybind11_add_holohub_module)
             ${MODULE_CPP_CMAKE_TARGET}
     )
 
+    # Conditionally link to the ABI config target if it exists (for HSDK >= 3.3)
+    set(pybind11_abi_details_msg "See https://github.com/pybind/pybind11/pull/2602 for details")
+    if(TARGET holoscan::pybind11)
+        message(STATUS "${target_name}: Linking against holoscan::pybind11 to disable strict ABI protection. ${pybind11_abi_details_msg}.")
+        target_link_libraries(${target_name} PRIVATE holoscan::pybind11)
+    else()
+        message(WARNING "${target_name}: holoscan::pybind11 target not found, using pybind11's default ABI protection. Are you using Holoscan SDK < 3.3? You might need to build ${target_name} with the same compiler and C++ library version as what this version of the Holoscan SDK was built with. ${pybind11_abi_details_msg}.")
```

Comment:
How should the user discover the compiler/stdlib used to build their HSDK version? Could we point to docs?

Reply:
Switched to message in cmake: https://github.com/nvidia-holoscan/holohub/pull/844/commits/2077f17c44a336b18fb0606136b2cc752a8155ac#diff-7331419371c0630f72747d4cd01eaa199222c61d33880dd09f09c15027154babR58

and extra info if `ImportError` occurs: https://github.com/nvidia-holoscan/holohub/pull/844/commits/2077f17c44a336b18fb0606136b2cc752a8155ac

The docs referenced will answer the questions you raised.